LibOS/shim/src/bookkeep/shim_vma.c

```diff
@@ -275,899 +298,538 @@ static inline void __set_comment (struct shim_vma * vma, const char * comment)
     vma->comment[len] = 0;
 }
 
-static int __bkeep_mmap (void * addr, uint64_t length,
-                         int prot, int flags,
+static int __bkeep_mmap (struct shim_vma * prev,
```

Comment:
I don't get the "attack scenario" you said since these are all in the same source file. "prev" is passed into __bkeep_mmap(), __bkeep_munmap(), and __bkeep_mprotect() because these functions can call each other and I want to avoid searching the list repeatedly. This is also for extensibility, in case in the future we want to switch to more advanced data structures such as rbtree.

Reply:
I didn't mean any "attack", rather an unnecessary complication to the interface which may induce bugs. But if there's a performance reason behind it then we can leave it as is.